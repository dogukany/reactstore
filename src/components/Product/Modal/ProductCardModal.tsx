import { BottomSheetBackdrop, BottomSheetModal } from "@gorhom/bottom-sheet";
import { Image, Text, View } from "react-native";

import { Ref, forwardRef, useCallback, useMemo } from "react";
import type { IProduct } from "../../../util/types/sliceTypes";
import styles from "./ProductCardModal.styles";

interface ProductCardModalProps {
  item: IProduct;
}

const ProductCardModal = forwardRef(
  (props: ProductCardModalProps, ref: Ref<BottomSheetModal>) => {
    const snapPoints = useMemo(() => ["50%"], []);
    const renderBackdrop = useCallback(
      (props) => (
        <BottomSheetBackdrop
          {...props}
          opacity={0.4}
          appearsOnIndex={0}
          disappearsOnIndex={-1}
        />
      ),
      []
    );

    return (
      <BottomSheetModal
        ref={ref}
        index={0}
        snapPoints={snapPoints}
        backdropComponent={renderBackdrop}
      >
        <View style={styles.contentContainer}>
          <Image
            source={{ uri: props.item.imageUrl }}
            style={styles.image}
            resizeMode="contain"
          />
          <Text style={styles.title}>{props.item.name}</Text>
          <Text style={styles.description}>{props.item.description}</Text>
        </View>
      </BottomSheetModal>
    );
  }
);

export default ProductCardModal;
