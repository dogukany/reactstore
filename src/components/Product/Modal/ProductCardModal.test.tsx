import { BottomSheetModalProvider } from "@gorhom/bottom-sheet";
import React from "react";
import uuid from "react-native-uuid";
import renderer from "react-test-renderer";
import ProductCardModal from "./ProductCardModal";

describe("ProductCardModal", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BottomSheetModalProvider>
          <ProductCardModal
            ref={React.createRef()}
            item={{
              id: uuid.v4(),
              name: "Test",
              price: 100,
              imageUrl: "https://via.placeholder.com/150",
              description: "Test",
              shippingMethod: "spaceship",
            }}
          />
        </BottomSheetModalProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
