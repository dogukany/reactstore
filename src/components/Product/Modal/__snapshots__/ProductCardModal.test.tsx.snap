// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardModal renders correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    [
      undefined,
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      {
        "bottom": 0,
        "overflow": "hidden",
        "top": 0,
      },
    ]
  }
/>
`;
